packages/x-components/tests/unit/main-scroll.spec.ts

```diff
@@ -166,6 +169,44 @@ describe('testing MainScroll component', () => {
         });
       });
 
+      it('restores the scroll with transitions enabled', () => {
+        const template = `
+        <MainScroll v-bind="{ threshold, margin, useWindow }">
+          <div data-test="main-scroll">
+            <FadeAndSlide>
```

Comment:
I would replace the `FadeAndSlide` with the `StaggeredFadeAndSlide`, which is the one that gives us more problems due to the delay.